### DISCUSSION TITLE
Making Select behave more like native select

### DISCUSSION THREAD

**Discussion by mayank99** [iTwin/iTwinUI-react#67] [Ideas]
`Select` cannot use native `select`, and using `input type=checkbox` or `input type=hidden` is bad. But it can still use a hidden `input` with the help of `opacity: 0`, `pointer-events: none` and `aria-hidden`.

This will allow passing native props like `required` and `disabled`, useful for browser form validation.

See this mui issue: https://github.com/mui-org/material-ui/issues/20402

---

Also maybe in the future we should support a "native Select" version (which does not have menu styling). This will have additional things like native events and mobile layout.

See MuiNativeSelect: https://material-ui.com/api/native-select/

**Comment by mayank99** [comment]
This is cool!
https://twitter.com/devongovett/status/1248306411508916224?s=20
https://github.com/adobe/react-spectrum

**Reply by bentleyvk** [reply]
I have read about this one and totally forgot 😀 It looks like a cool lib with a cool idea to have component logic separate from the component itself. I think if you really don't want to spend a lot of time on your own design system, you can use adobe-spectrum provided hooks and just apply your own styling/structure.

**Comment by mayank99** [comment]
This is really interesting: https://open-ui.org/components/select

Open UI is proposing a standardized way of customizing these hard-to-style native inputs. It might get implemented in browsers in the future.

**Reply by bentleyvk** [reply]
So like after 2 years we could use this 😀 Nice finding!
